src/App.js

```diff
@@ -89,16 +89,18 @@ function App() {
       return { score: bestScore, move: move };
     }
   }
+
   // This useEffect hook runs whenever the state of squares or result changes.
   useEffect(() => {
     const isComputerTurn = squares.filter(square => square !== null).length % 2 === 1;
-    const result = calculateResult(squares);
+    const currentResult = calculateResult(squares);
```

Comment:
so this feels like a weird way to set `currentResult`. Since you already had the state variable `result` and then `setResult`, you could just set the result in the `calculateResult` function and then keep using `result` in this `useState`